bundles/org.openhab.binding.dreamscreen/src/main/java/org/openhab/binding/dreamscreen/internal/DreamScreenDatagramServer.java

```diff
@@ -0,0 +1,272 @@
+
+                if (isValidStateMsg(buf, off, len) && isDreamScreen(buf, off, len)) {
+                    logger.debug("Received DreamScreen message from {}", data.getAddress());
+                    refreshDreamScreen(buf, off, len, data.getAddress());
+                }
+            } catch (IOException ioe) {
+                logger.error("Error receiving DreamScreen data", ioe);
+            }
+        }
+    }
+
+    void setHostAddress(@Nullable String value) {
+        try {
+            this.hostAddress = InetAddress.getByName(value);
+        } catch (UnknownHostException e) {
+            logger.error("Cannot set host address to {}", value, e);
+        }
+
+        try {
+            if (this.refreshing != null) {
+                stopRefreshing();
+                ensureRefreshing();
+            }
+        } catch (IOException e) {
+            logger.error("Cannot restart DreamScreen refresh on new host address {}", value, e);
+        }
+    }
+
+    private boolean isValidStateMsg(final byte[] buf, int off, int len) {
+        if (isValidMsg(buf, off, len)) {
+            final int upperCommand = buf[off + 4];
+            final int lowerCommand = buf[off + 5];
+
+            return upperCommand == 0x01 && lowerCommand == 0x0A;
+        }
+        return false;
+    }
+
+    private boolean isValidMsg(final byte[] buf, int off, int len) {
+        if (len > 6 && buf[off] == (byte) 0xFC) {
+            final int msgLen = buf[off + 1] & 0xFF;
+            if (msgLen + 2 > len) {
+                return false; // invalid length
+            } else if (buf[off + msgLen + 1] != calcCRC8(buf, off)) {
+                return false; // invalid crc
+            }
+            return true;
+        }
+        return false; // message not long enough
+    }
+
+    private boolean isDreamScreen(final byte[] buf, int off, int len) {
+        final int msgLen = buf[off + 1] & 0xFF;
+        final int productId = buf[off + msgLen];
+
+        return msgLen > 77 && productId > 0 && productId <= 2;
+    }
+
+    @SuppressWarnings("null")
+    private void refreshDreamScreen(final byte[] buf, int off, int len, InetAddress address) {
+        final String name = new String(buf, off + 6, 16, StandardCharsets.UTF_8).trim();
+        final DreamScreenHandler dreamScreen = this.dreamScreens.get(name);
+
+        if (dreamScreen != null) {
+            dreamScreen.refreshState(buf, off, len, address);
+        }
+    }
+
+    void send(int group, int commandUpper, int commandLower, byte[] payload, @Nullable InetAddress address)
+            throws IOException {
+        send(group, 0b00010001, commandUpper, commandLower, payload, address);
+    }
+
+    void send(int group, int flags, int commandUpper, int commandLower, byte[] payload, @Nullable InetAddress address)
+            throws IOException {
+        final DatagramSocket socket = new DatagramSocket();
```

Comment:
You can use try-with-resources and omit the close() call.
```suggestion
        try (final DatagramSocket socket = new DatagramSocket()) {
```

Reply:
changed